universum/analyzers/utils.py

```diff
@@ -4,37 +4,88 @@
 import glob
 import subprocess
 
-from typing import Callable, List, Optional
+from typing import Any, Callable, List, Optional, Tuple
 from typing_extensions import TypedDict
 
-ReportData = TypedDict('ReportData', {'path': str, 'message': str, 'symbol': str, 'line': int})
-
+from universum.lib.ci_exception import CiException
 
-def report_parsed_outcome(cmd: Optional[List[str]],
-                          parse_function: Callable[[str], List[ReportData]],
-                          out_file: str) -> int:
-    if cmd:
-        result = subprocess.run(cmd, universal_newlines=True,  # pylint: disable=subprocess-run-check
-                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        if result.stderr and not result.stdout:
-            sys.stderr.write(result.stderr)
-            return result.returncode
-        data_to_parse = result.stdout
-    else:
-        data_to_parse = ''  # nothing to parse from stdout
+ReportData = TypedDict('ReportData', {'path': str, 'message': str, 'symbol': str, 'line': int})
 
-    try:
-        parsed_issues = parse_function(data_to_parse)
-    except Exception as e:
-        sys.stderr.write("Error encountered while parsing output:\n")
-        sys.stderr.write(str(e))
-        return 2
 
-    output_report_to_file(parsed_issues, out_file)
-    return 1 if parsed_issues else 0
+class AnalyzerException(CiException):
+    def __init__(self, code: int = 2, message: Optional[str] = None):
+        self.code: int = code
+        self.message: Optional[str] = message
+
+
+def analyzer(parser: argparse.ArgumentParser):
```

Comment:
If I understand the idea of analyzer and sys_exit decorators correctly, they define the interface between analyzers and CodeReportCollector. Please document this idea (the purpose) in the code.

Reply:
yes, this function serves to codify the interface between analyzer and reporter